app/integration/runtime/src/main/java/io/syndesis/integration/runtime/util/JsonSimplePredicate.java

```diff
@@ -16,101 +16,158 @@
 package io.syndesis.integration.runtime.util;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.MapLikeType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+
 import org.apache.camel.CamelContext;
 import org.apache.camel.Exchange;
+import org.apache.camel.Message;
 import org.apache.camel.Predicate;
 import org.apache.camel.spi.Language;
 import org.apache.camel.spi.Registry;
 import org.apache.camel.util.ExchangeHelper;
 import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Predicate which tries to convert a JSON message to a map first before
  * applying
  */
-public class JsonSimplePredicate implements Predicate {
+public final class JsonSimplePredicate implements Predicate {
+    private static final MapLikeType GENERIC_TYPE = TypeFactory.defaultInstance().constructMapLikeType(HashMap.class, String.class,
+        Object.class);
+
+    private static final Logger LOG = LoggerFactory.getLogger(JsonSimplePredicate.class);
+
     private static final ObjectMapper MAPPER = new ObjectMapper();
```

Comment:
Do we need also to support jdk8 date/time & so on ?

Reply:
Perhaps, I'm not sure if simple filter supports dates though...